pkg/server/server.go

```diff
@@ -0,0 +1,629 @@
+	}
+	//Print to console for logging
+	fmt.Printf("Created Secret: %v\n", secret.GetName())
+
+	js, err := json.Marshal(createdNs)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error marshalling namespace: %v\n", err)
+	}
+	w.WriteHeader(201)
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(js)
+}
+
+//getEnvironment returns a kubernetes namespace matching the given environmentGroupID and environmentName
+func getEnvironment(w http.ResponseWriter, r *http.Request) {
+	pathVars := mux.Vars(r)
+
+	labelSelector, err := labels.Parse("Group=" + pathVars["environmentGroupID"])
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error creating label selector in getEnvironment: %v\n", err)
+		return
+	}
+
+	nsList, err := client.Namespaces().List(api.ListOptions{
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error in getEnvironment: %v\n", err)
+		return
+	}
+	//Flag indicating there is at least one value matching that name
+	flag := false
+
+	for _, value := range nsList.Items {
+		if value.GetName() == pathVars["environmentGroupID"]+"-"+pathVars["environment"] {
+			flag = true
+			js, err := json.Marshal(value)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				fmt.Printf("Error marshalling namespace: %v\n", err)
+			}
+			w.Header().Set("Content-Type", "application/json")
+			w.Write(js)
+			fmt.Printf("Got Namespace: %v\n", value.GetName())
+		}
+	}
+	if flag != true {
+		w.WriteHeader(404)
+		fmt.Fprintf(w, "Environment not found")
+	}
+}
+
+func updateEnvironment(w http.ResponseWriter, r *http.Request) {
+	pathVars := mux.Vars(r)
+
+	//Struct to put JSON into
+	type environmentPost struct {
+		Secret string `json:"secret"`
+	}
+	//Decode passed JSON body
+	decoder := json.NewDecoder(r.Body)
+	var tempJSON environmentPost
+	err := decoder.Decode(&tempJSON)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error decoding JSON Body: %v\n", err)
+		return
+	}
+
+	//Check if there is a secret named ingress in the given environment
+	getSecret, err := client.Secrets(pathVars["environmentGroupID"] + "-" + pathVars["environment"]).Get("ingress")
+	if err != nil {
+		test := errors.New("secrets \"ingress\" not found")
+		if err.Error() == test.Error() {
+			//Create secret
+			tempSecret := api.Secret{
+				ObjectMeta: api.ObjectMeta{
+					Name: "ingress",
+				},
+				Data: map[string][]byte{
+					"api-key": []byte(tempJSON.Secret),
+				},
+				Type: "Opaque",
+			}
+			secret, err := client.Secrets(pathVars["environmentGroupID"] + "-" + pathVars["environment"]).Create(&tempSecret)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				fmt.Printf("Error creating secret: %v\n", err)
+			}
+
+			w.WriteHeader(200)
```

Comment:
I wonder if we should be using `204` for successful requests that have no response content?  Not stopping this merge, just thinking aloud.